src/GitHub.App/ViewModels/GitHubPane/PullRequestCheckViewModel.cs

```diff
@@ -53,8 +55,55 @@ public static IEnumerable<IPullRequestCheckViewModel> Build(IViewViewModelFactor
                     : AvatarProvider.CreateBitmapImage(DefaultAvatar);
 
                 return pullRequestCheckViewModel;
-
             }) ?? new PullRequestCheckViewModel[0];
+
+            var checks = pullRequest.CheckSuites?.SelectMany(model => model.CheckRuns)
+                .Select(model =>
+                {
+                    PullRequestCheckStatus checkStatus;
+                    switch (model.Status)
+                    {
+                        case CheckStatusState.Requested:
+                        case CheckStatusState.Queued:
+                        case CheckStatusState.InProgress:
+                            checkStatus = PullRequestCheckStatus.Pending;
+                            break;
+
+                        case CheckStatusState.Completed:
+                            switch (model.Conclusion)
+                            {
+                                case CheckConclusionState.Success:
+                                    checkStatus = PullRequestCheckStatus.Success;
+                                    break;
+
+                                case CheckConclusionState.ActionRequired:
+                                case CheckConclusionState.TimedOut:
+                                case CheckConclusionState.Cancelled:
+                                case CheckConclusionState.Failure:
+                                case CheckConclusionState.Neutral:
+                                    checkStatus = PullRequestCheckStatus.Failure;
+                                    break;
+
+                                default:
+                                    throw new ArgumentOutOfRangeException();
+                            }
+
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException();
+                    }
+
+                    var pullRequestCheckViewModel = (PullRequestCheckViewModel)viewViewModelFactory.CreateViewModel<IPullRequestCheckViewModel>();
+                    pullRequestCheckViewModel.CheckType = PullRequestCheckType.ChecksApi;
+                    pullRequestCheckViewModel.Title = model.Name;
+                    pullRequestCheckViewModel.Description = model.Summary;
+                    pullRequestCheckViewModel.Status = checkStatus;
+                    pullRequestCheckViewModel.DetailsUrl = new Uri(model.DetailsUrl);
+
+                    return pullRequestCheckViewModel;
+                }) ?? new PullRequestCheckViewModel[0];
+
+            return statuses.Union(checks).OrderBy(model => model.Title);
```

Comment:
Is `.Union` the best operator to use here? It will do equality checks for each item, even though the items will never be equal. Not a big deal but I would have said that `.Concat` would be a better choice here.

Reply:
Oh. I never knew that...